src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -479,268 +578,513 @@ public void LastIndexOf()
                 (b, v, i, c, eq) => b.LastIndexOf(v, i, c, eq));
         }
 
+        [Theory]
+        [MemberData(nameof(Int32EnumerableData))]
+        public void ContainsInt32(IEnumerable<int> source)
+        {
+            var array = source.ToImmutableArray();
+
+            if (source.Any(i => i >= 0))
+            {
+                int contained = Enumerable.Range(0, int.MaxValue).First(i => source.Contains(i));
+                Assert.True(array.Contains(contained));
+                Assert.True(((ICollection<int>)array).Contains(contained));
+            }
+
+            int notContained = Enumerable.Range(0, int.MaxValue).First(i => !source.Contains(i));
+            Assert.False(array.Contains(notContained));
+            Assert.False(((ICollection<int>)array).Contains(notContained));
+        }
+
+        [Theory]
+        [MemberData(nameof(ContainsNullData))]
+        public void ContainsNull<T>(IEnumerable<T> source, T dummy) where T : class
+        {
+            // Remove the dummy parameters once https://github.com/xunit/xunit/pull/965 makes it into corefx.
+
+            bool expected = source.Contains(null, EqualityComparer<T>.Default);
+            Assert.Equal(expected, source.ToImmutableArray().Contains(null));
+        }
+
+        public static IEnumerable<object[]> ContainsNullData()
+        {
+            yield return new object[] { s_oneElementRefType, new GenericParameterHelper() };
+            yield return new object[] { s_twoElementRefTypeWithNull, string.Empty };
+            yield return new object[] { ImmutableArray.Create(new object()), new object() };
+        }
+
         [Fact]
-        public void Contains()
+        public void ContainsInvalid()
         {
             Assert.Throws<NullReferenceException>(() => s_emptyDefault.Contains(0));
-            Assert.False(s_empty.Contains(0));
-            Assert.True(s_oneElement.Contains(1));
-            Assert.False(s_oneElement.Contains(2));
-            Assert.True(s_manyElements.Contains(3));
-            Assert.False(s_oneElementRefType.Contains(null));
-            Assert.True(s_twoElementRefTypeWithNull.Contains(null));
         }
 
-        [Fact]
-        public void ContainsEqualityComparer()
+        // TODO: This was actually calling the Enumerable extension method.
+        // There exists no such method on ImmutableArray.
+        // Also, there are no tests for a null equality comparer.
+
+        [Theory]
+        [MemberData(nameof(ContainsEqualityComparerData))]
+        public void ContainsEqualityComparer<T>(IEnumerable<T> source, T value, IEqualityComparer<T> comparer)
```

Comment:
After cleaning up this test I found out this was actually calling a method in LINQ, so it isn't really relevant to ImmutableArray (aside from testing GetEnumerator, which is covered below). Should I remove this?